CMakeLists.txt

```diff
@@ -42,6 +42,8 @@ option(BUILD_DAEMON "Build bitcoind executable." ON)
 option(ASM "Use assembly routines." ON)
 cmake_dependent_option(CXX20 "Enable compilation in C++20 mode." OFF "NOT MSVC" ON)
 option(THREADLOCAL "Enable features that depend on the C++ thread_local keyword (currently just thread names in debug logs)." ON)
+include(TristateOption)
+tristate_option(CCACHE "Use ccache for compiling." "if ccache is found." AUTO)
```

Comment:
From @fanquake on #12:

> Yea. We certainly wont be adopting this tri-state logic/machinery into our CMake build system long-term/at-merge-time. I'd consider it a regression to reimplement this Autotools behaviour. As for disabling certain "features" by default, yes, I think that is what we are going to do. It continues to make less & less sense to compile in stuff like this by default, let alone auto-opt-in builders to it, based on what happens to be installed on their systems at the time of compilation.

At this point I agree with that and I'd prefer to just never start with this tri-state stuff as it's quite not-cmake-like. But I realize I originally asked for 1:1 feature parity with autotools for the sake of review, so it's not really fair to ask for that change now.

How about a comment above with a loud warning that says "These tri-state options will be removed and most features will become opt-in by default before merging into master" ?

Reply:
[Updated](https://github.com/hebasto/bitcoin/pull/13#issuecomment-1507244320).